### PR TITLE
fix: validate deviceClass name to be usable in storageClass

### PR DESCRIPTION
**Pull request by leelavg**
Fixes: #40

Signed-off-by: Leela Venkaiah G <lgangava@redhat.com>

### CODE REVIEW

**Review by jmolmo — APPROVED** [review]

File: api/v1alpha1/lvmcluster_types.go [hunk 1]
```diff
@@ -35,6 +35,11 @@ type LVMClusterSpec struct {
 
 type DeviceClass struct {
 	// Name of the class, the VG and possibly the storageclass.
+	// Validations to confirm that this field can be used as metadata.name field in storageclass
+	// ref: https://github.com/kubernetes/apimachinery/blob/de7147/pkg/util/validation/validation.go#L209
+	// +kubebuilder:validation:MaxLength=245
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Pattern="^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$"
 	Name string `json:"name,omitempty"`
```
Comment (nbalacha) [inline api/v1alpha1/lvmcluster_types.go:43]: I don't think this should have a "." in the name. The first part - "^[a-z0-9]([-a-z0-9]*[a-z0-9])?" -  is sufficient.

Comment (nbalacha) [reply api/v1alpha1/lvmcluster_types.go:43]: It should follow this [pattern](https://github.com/kubernetes/apimachinery/blob/de7147de41c9331e4697399f48564f1d970cc01e/pkg/util/validation/validation.go#L178):
const dns1123LabelFmt string = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"

Comment (leelavg) [reply api/v1alpha1/lvmcluster_types.go:43]: ack, I'll drop the subdomain format.

**Review by nbalacha — APPROVED** [review]
From man lvm: The valid characters for VG and LV names are: a-z A-Z 0-9 + _ . -
However, the operator will restrict the name to characters such that it can be used to generate valid kubernetes resource names

**Pull request merged**